### PR TITLE
Move docserv code into modules

### PR DESCRIPTION
**Pull request by svenseeberg**
* Set entry point in setup.py

### CODE REVIEW

File: src/docserv/bih.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+                self.remote_repo), thread_id, gitLocks, gitLocksLock)
+            self.prepare_repo(thread_id)
+            self.get_commit_hash()
+        else:
+            self.initialized = False
+        return
+
+    def cleanup(self):
+        """
+        Remove temporary files when build failes or all deliverables
```
Comment (tomschr) [inline src/docserv/bih.py]: s/failes/fails/

File: src/docserv/bih.py [hunk 2]
```diff
@@ -0,0 +1,389 @@
+import json
```
Comment (tomschr) [inline src/docserv/bih.py]: Actually, it's for the `__init__.py` file (GitHub doesn't allow me to add a comment on an empty file, grrr):

I would add the following information:

* `__version__` contains the version of docserv²
* `__author__` your name. :wink:

File: setup.py [hunk 3]
```diff
@@ -19,12 +19,24 @@
...
+from glob import glob
 
 setup(
     name="docserv",
     version="0.9.16",
-    packages=find_packages(),
-    scripts=['bin/docserv', 'bin/docserv-stitch', 'bin/docserv-dirs', 'bin/docserv-createconfig'],
+    packages=find_packages('src'),
+    package_dir={'': 'src'},
+    py_modules=[splitext(basename(path))[0] for path in glob('src/*.py')],
```
Comment (tomschr) [inline setup.py]: Not sure if you really need `py_modules` here...

File: src/docserv/bih.py [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+import random
+import hashlib
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from socketserver import ThreadingMixIn
+from xml.etree import ElementTree, cElementTree
+from email.mime.text import MIMEText
+
+from docserv.deliverable import Deliverable
+from docserv.repolock import RepoLock
+from docserv.functions import resource_to_filename, mail
```
Comment (tomschr) [inline src/docserv/bih.py]: It may be a nice addition: you could use relative imports. The only thing you need to change is to remove `docserv`.

Comment (svenseeberg) [reply src/docserv/bih.py]: I tried, but then it fails in develop mode of setuptools.

Comment (tomschr) [reply src/docserv/bih.py]: Never mind. Just one of my crazy ideas. :laughing:

**Pull request merged**